agilent_infiniium_scope.py

```diff
@@ -72,319 +72,350 @@ def channel_infoNONEXISTANT(self, c, channel):
+    @setting(117, channel='i', position='v', returns=['v'])
+    def position(self, c, channel, position=None):
         """Get or set the voltage at the center of the screen
         """
         dev = self.selectedDevice(c)
         if position is None:
-            resp = yield dev.query('CHAN%d:OFFS?' %channel)
+            resp = yield dev.query('CHAN{}:OFFS?'.format(channel))
         else:
-            yield dev.write(('CHAN%d:OFFS %f') %(channel,position))
-            resp = yield dev.query('CHAN%d:OFFS?' %channel)
+            yield dev.write('CHAN{}:OFFS {}'.format(channel, position))
+            resp = yield dev.query('CHAN{}:OFFS?'.format(channel))
         position = float(resp)
         returnValue(position)
         
-    @setting(118, mode =  ['s','i','b'], returns = ['s'])
-    def averagemode(self, c, mode = None):
+    @setting(118, mode=['s', 'i', 'b'], returns=['s'])
+    def averagemode(self, c, mode=None):
         """Get or set acquisition mode
         """
         dev = self.selectedDevice(c)
-        if mode is None:
-            resp = yield dev.query('ACQ:AVER?')
-        else:
+        if mode is not None:
             if mode is not None:
                 if isinstance(mode, str):
-                    mode = {'ON':1,'OFF':0}[mode]
+                    mode = {'ON': 1, 'OFF': 0}[mode]
                 elif isinstance(mode, bool):
                     mode = int(mode)
                 elif isinstance(mode, int):
                     pass       
-            yield dev.write('ACQ:AVER '+str(mode))
-            resp = yield dev.query('ACQ:AVER?')
+            yield dev.write('ACQ:AVER {}'.format(mode))
+        resp = yield dev.query('ACQ:AVER?')
         returnValue(resp) 
         
-    @setting(119, navg = 'i', returns = ['i'])
-    def numavg(self, c, navg = None):
+    @setting(119, navg='i', returns=['i'])
+    def numavg(self, c, navg=None):
         """Get or set number of averages
         """
         dev = self.selectedDevice(c)
         if navg is None:
             resp = yield dev.query('ACQ:COUN?')
-        else:    
-            yield dev.write('ACQ:COUN %d' %navg)
+        else:
+            yield dev.write('ACQ:COUN {}'.format(navg))
             resp = yield dev.query('ACQ:COUN?')
         navg_out = int(resp)
-        returnValue(navg_out)         
+        returnValue(navg_out)
+
+    @setting(131, channel='i', level='v', returns='v{level}')
+    def trigger_at(self, c, channel, level=None):
+        """Get or set the trigger source and voltage for edge mode triggering.
 
-    @setting(131, channel = 'i', level = 'v', returns = 'v{level}')
-    def trigger_at(self, c, channel, level = None):
-        """Get or set the trigger source and the trigger voltage for edge mode triggering.
-        Channel must be one of 0 (AUX), 1, 2, 3, 4.
+        Channel must be one of 0 (AUX), 1, 2, 3, 4, or 5 (LINE).
+        If trigger source is 5 (LINE) the level will be ignored
```

Comment:
For these scopes at least.  I probably realized that by trial and error, but here is it confirmed from the manual:

http://www.keysight.com/upload/cmc_upload/All/9000_series_prog_ref.pdf#page=914